test_tutorial_base.sh is indicating a build failure in the tutorials

**Describe the bug**
I tried to run `test_tutorial_base.sh` and I expected it to pass instead it failed at build time.

**GEOPM version**
`d855769d`

**Expected behavior**
The test pass and have an RC of 0.

**Actual behavior**
```
$ ./integration/test/test_tutorial_base.sh
...
mpiicc -I/home/test/.local/tmp/geopm-tutorial-test.k9hhtgjM/build//home/test/build/geopm/include -qopenmp -DTUTORIAL_ENABLE_MKL -D_GNU_SOURCE -std=c99 -xAVX  -c tutorial_region_prof.c -o tutorial_region_prof.o
mpiicc tutorial_5.o tutorial_region.o tutorial_region_prof.o -o tutorial_5 -L/home/test/.local/tmp/geopm-tutorial-test.k9hhtgjM/build//home/test/build/geopm/lib -qopenmp -lm -lrt -mkl -xAVX  -lgeopm -lstdc++
icpc -I/home/test/.local/tmp/geopm-tutorial-test.k9hhtgjM/build//home/test/build/geopm/include -qopenmp -DTUTORIAL_ENABLE_MKL -D_GNU_SOURCE -std=c++11 -xAVX  -I ./agent -fPIC -shared agent/ExampleAgent.cpp -o agent/libgeopmagent_example_agent.so.0.0.0
/home/test/.local/tmp/geopm-tutorial-test.k9hhtgjM/build//home/test/build/geopm/include/geopm/Helper.hpp(25): error: object of abstract class type "ExampleAgent" is not allowed:
            pure virtual function "geopm::Agent::enforce_policy" has no overrider
          return std::unique_ptr<Type>(new Type(std::forward<Args>(args)...));
                                           ^
          detected during instantiation of "std::unique_ptr<Type, std::default_delete<Type>> geopm::make_unique<Type,Args...>(Args &&...) [with Type=ExampleAgent, Args=<>]" at line 240 of "agent/ExampleAgent.cpp"

compilation aborted for agent/ExampleAgent.cpp (code 2)
make: *** [Makefile:72: agent/libgeopmagent_example_agent.so.0.0.0] Error 2
```

Running `tutorial_build_intel.sh` out of the `tutorials/` directory also exhibits this build error.

New behavior introduced with PR #2476.

--- a/tutorial/agent/ExampleAgent.cpp
+++ b/tutorial/agent/ExampleAgent.cpp
@@ -228,6 +228,11 @@ std::vector<std::function<std::string(double)> > ExampleAgent::trace_formats(voi
             };
 }
 
+void ExampleAgent::enforce_policy(const std::vector<double> &policy) const
+{
+
+}
+
 // Name used for registration with the Agent factory
 std::string ExampleAgent::plugin_name(void)
 {

--- a/tutorial/agent/ExampleAgent.hpp
+++ b/tutorial/agent/ExampleAgent.hpp
@@ -41,6 +41,7 @@ class ExampleAgent : public geopm::Agent
         std::vector<std::string> trace_names(void) const override;
         void trace_values(std::vector<double> &values) override;
         std::vector<std::function<std::string(double)> > trace_formats(void) const override;
+        void enforce_policy(const std::vector<double> &policy) const override;
 
         static std::string plugin_name(void);
         static std::unique_ptr<geopm::Agent> make_plugin(void);

